analytics/client.py

```diff
@@ -53,8 +57,12 @@ def __init__(self, write_key=None, debug=False, max_queue_size=10000,
             self.consumer = S3Consumer(self.queue, write_key, endpoint=endpoint,
                                        on_error=on_error, upload_size=upload_size,
                                        key_decorator=key_decorator)
+        elif transport == 's3_delete_first':
+            self.consumer = S3Consumer(self.queue, write_key, endpoint=endpoint,
+                                       on_error=on_error, upload_size=upload_size,
+                                       key_decorator=key_decorator, delete_first=True)
```

Comment:
I agree with @pavel-popov, let's move the `delete_first` as a parameter of the `_init_` constructor and set it to `False` by default for `s3`.

Reply:
That was my first choice, but then I would have to change the signature of the Client constructor by adding a param that would only be related to the S3Consumer, not the standard Consumer, and that could be misleading.

This is why I preferred to use a new transport type, instead.

Do you still prefer to move it to the Client constructor?

Reply:
I'm fine with the way you did it. It could be nicer/simpler to have it the other way, but I don't think it's a blocker. For me the PR is ok.